contracts/SpokePool.sol

```diff
@@ -675,6 +690,83 @@ abstract contract SpokePool is
         _emitFillRelay(relayExecution, fillAmountPreFees);
     }
 
+    /**
+     * @notice Caller signals to the system that they want a refund on this chain, which they set as the
+     * `repaymentChainId` on the original fillRelay() call on the `destinationChainId`. An observer should be
+     * be able to 1-to-1 match the emitted RefundRequested event with the FilledRelay event on the `destinationChainId`.
+     * @dev This function could be used to artificially inflate the `fillCounter`, allowing the caller to "frontrun"
+     * and cancel pending fills in the mempool. This would in the worst case censor fills at the cost of the caller's
+     * gas costs. We don't view this as a major issue as the fill can be resubmitted and obtain the same incentive,
+     * since incentives are based on validated refunds and would ignore these censoring attempts. This is no
+     * different from calling `fillRelay` and setting msg.sender = recipient.
+     * @dev Caller needs to pass in `fillBlock` that the FilledRelay event was emitted on the `destinationChainId`.
+     * This is to make it hard to request a refund before a fill has been mined and to make lookups of the original
+     * fill as simple as possible.
+     * @param refundToken This chain's token equivalent for original fill destination token.
+     * @param amount Original deposit amount.
+     * @param originChainId Original origin chain ID.
+     * @param destinationChainId Original destination chain ID.
+     * @param realizedLpFeePct Original realized LP fee %.
+     * @param depositId Original deposit ID.
+     * @param maxCount Max count to protect the refund recipient from frontrunning.
+     */
+    function requestRefund(
+        address refundToken,
+        uint256 amount,
+        uint256 originChainId,
+        uint256 destinationChainId,
+        int64 realizedLpFeePct,
+        uint32 depositId,
+        uint256 fillBlock,
+        uint256 maxCount
+    ) external nonReentrant {
+        // Prevent unrealistic amounts from increasing fill counter too high.
+        require(amount <= MAX_TRANSFER_SIZE, "Amount too large");
+
+        // This allows the caller to add in frontrunning protection for quote validity.
+        require(fillCounter[refundToken] <= maxCount, "Above max count");
+
+        // Track duplicate refund requests.
+        bytes32 refundHash = keccak256(
+            abi.encode(msg.sender, refundToken, amount, originChainId, realizedLpFeePct, depositId, fillBlock)
```

Comment:
Do you need to add the destinationChainId to this abi.encode to ensure that refundHash doesn't match for different destinationChainIds?

Reply:
Great call, done